examples/NSIDC/cmr_auth.py

```diff
@@ -0,0 +1,66 @@
+from copy import deepcopy
+from typing import Dict, Any
+
+from requests import session, exceptions
+from requests.auth import HTTPBasicAuth
+
+
+class CMRAuth(object):
+    """
+    base class to perform authentication operations for NASA CMR
+    """
+    CMR_OPS = "https://cmr.earthdata.nasa.gov/legacy-services/rest/tokens"
+    CMR_UAT = "https://cmr.uat.earthdata.nasa.gov/legacy-services/rest/tokens"
+    CMR_SIT = "https://cmr.sit.earthdata.nasa.gov/legacy-services/rest/tokens"
```

Comment:
Question: are CMR_UAT and CMR_SIT accessible to everyone? Even if they are accessible, do we want the general public hitting these endpoints? Should these URLs be here?

Reply:
They are public [CMR endpoints](https://wiki.earthdata.nasa.gov/display/CMR/CMR+Client+Partner+User+Guide#CMRClientPartnerUserGuide-CreatingaToken). We could only include prod as I don't see a use case where someone wants to do something with the test environments.